File: src/algorithm_java/leetcode/Problem338.java
package algorithm.leetcode;

public class Problem338 {

    public int[] countBits(int num) {
        int[] counts = new int[num + 1];
        for (int i = 1; i <= num; i++) {
            counts[i] = counts[i & (i - 1)] + 1;
        }
        return counts;
    }

}

File: src/algorithm_cpp/problem338.cpp
class Solution {
public:
    vector<int> countBits(int num) {
	vector<int> rt(num + 1, 0);
	for (int i = 1; i <= num; i++) {
	    rt[i] = rt[i >> 1] + (i & 1);
	}
	return rt;
    }
};
